# HG changeset patch
# User StrikerTheHedgefox <mossj32@gmail.com>
# Date 1743762097 10800
#      Fri Apr 04 07:21:37 2025 -0300
# Node ID 7f12f05ea1a8d89c2724daea329610cdba0b1513
# Parent  57041369f51ae8b3502c6a321cafff2aa44337aa
# EXP-Topic strikers_stuff
Fix truncation of translation index in decals.

diff -r 57041369f51a -r 7f12f05ea1a8 docs/zandronum-history.txt
--- a/docs/zandronum-history.txt	Fri Apr 04 07:13:33 2025 -0300
+++ b/docs/zandronum-history.txt	Fri Apr 04 07:21:37 2025 -0300
@@ -18,6 +18,7 @@
 +	- Added A_JumpTo codepointer, which jumps to a state instantly & with prediction. Useful for jumping to an ancestor actor's state without the issues A_Jump with a 256 chance has. [StrikerTheHedgefox]
 +	- Backported A_WeaponOffset from ZDoom Revision 115dbd0, with alterations necessary for Zandronum's current codebase. [StrikerTheHedgefox]
 -	- Fix a crash that can occur when using the "colors" DECALDEF token, when the decal's image was in the "graphics" folder. [StrikerTheHedgefox]
+-	- Fix truncation of translation index in decals. [StrikerTheHedgefox]
 
 3.2
 ---
diff -r 57041369f51a -r 7f12f05ea1a8 src/decallib.cpp
--- a/src/decallib.cpp	Fri Apr 04 07:13:33 2025 -0300
+++ b/src/decallib.cpp	Fri Apr 04 07:21:37 2025 -0300
@@ -96,7 +96,7 @@
 
 	DWORD StartColor, EndColor;
 	FTranslation *Next;
-	WORD Index;
+	DWORD Index;
 };
 
 struct FDecalAnimator
@@ -1106,7 +1106,7 @@
 		table[i] = ColorMatcher.Pick (ri >> 24, gi >> 24, bi >> 24);
 	}
 	table[0] = table[1];
-	Index = (WORD)TRANSLATION(TRANSLATION_Decals, tablei >> 8);
+	Index = (DWORD)TRANSLATION(TRANSLATION_Decals, tablei >> 8);
 }
 
 FDecalLib::FTranslation *FDecalLib::FTranslation::LocateTranslation (DWORD start, DWORD end)
